### PR TITLE
chore: add release template & update contributing

### PR DESCRIPTION
**Pull request by schristoff-du**

### CODE REVIEW

**Review by lucasrod16 — APPROVED** [review]
LGTM

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+📅 Target dates:
+
+Release Manager: 
+
+Release checklist:
```
Comment (lucasrod16) [inline .github/ISSUE_TEMPLATE/release.md:6]: We should add merging the `goreleaser` PR in the [homebrew-tap](https://github.com/defenseunicorns/homebrew-tap) repo as a step in the release checklist

**Review by lucasrod16 — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/release.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+🚨 Update the release version variable to the version
+you are releasing. 
+
+[ ] `export RELEASE_VERSION=v0.33.1`
+
+[ ] `git tag -sa $RELEASE_VERSION -m "$RELEASE_VERSION" `
```
Comment (lucasrod16) [inline .github/ISSUE_TEMPLATE/release.md:12]: Would be good to note that the tag has to be pushed to remote:

```shell
git push origin "$RELEASE_VERSION"
```

**Review by lucasrod16 — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/release.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+[ ] Send release update in the #zarf Kubernetes Channel
```
Comment (AustinAbro321) [inline .github/ISSUE_TEMPLATE/release.md:14]: We should add an item to this checklist to merge the PR goreleaser creates on the [homebrew-tap repo](https://github.com/defenseunicorns/homebrew-tap)

**Review by AustinAbro321 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**